curated_transformers/pipe.py

```diff
@@ -280,28 +290,36 @@ def update(
         validate_examples(examples, "Transformer.update")
         docs = [eg.predicted for eg in examples]
         set_dropout_rate(self.model, drop)
-        tokvecs, bp_tokvecs = self.model.begin_update(docs)
-        d_tokvecs = [
-            Ragged(self.model.ops.alloc_f(t2v.dataXd.shape), t2v.lengths)
-            for t2v in tokvecs
-        ]
         losses.setdefault(self.name, 0.0)
 
+        if not self.frozen:
```

Comment:
Since we have to handle both cases anyway, maybe switch them around, so that we can have

```
if self.frozen:
```